### DISCUSSION TITLE
Texture upload/re-usage question

### DISCUSSION THREAD

**Discussion by demiantres** [mapsforge/vtm#865] [General]
Hi,

I am trying to understand how textures are handled by VTM.

1.) Is it correct that for each marker an OpenGL texture is created [in the cluster examples], even when the markers share the same underlying bitmap?

2.) Is it correct that the location texture renderer uploads a new texture each time the location/rotation changes? Wouldn't it be much better to re-use the texture and just apply the new matrices? I am not entirely sure how this can be done because the example uses a SymbolBucket. So, is re-using possible with the current system, or has the SymbolBucket to be modified?

Thanks,

**Comment by devemux86** [comment]
1. The cluster sample is just an experiment provided by the users.

2. The location renderers are just examples for how location "could" be displayed.
Anyone can extend or rewrite a different custom implementation.

**Reply by demiantres** [reply]
Thanks for your answer. I am aware that both are examples. But I am trying to understand/improve them. Do you have a hint/example in the project how I could re-use the texture with a SymbolBucket? In other words: I want to change the symbol's parameters (rotation, billboard, ...) but reuse the existing uploaded texture.
